neuro_extras/main.py

```diff
@@ -221,6 +252,182 @@ def image() -> None:
+        raise ValueError(
+            "This command can't be used to copy data between two storage locations"
+        )
+
+    if UrlType.STORAGE in (source_url_type, destination_url_type):
+        async with neuro_api.get() as client:
+            data_copier = DataCopier(client)
+            if source_url_type == UrlType.STORAGE:
+                job = await data_copier.launch(
+                    storage_uri=source_url,
+                    src_uri=URL("/var/storage"),
+                    dst_uri=destination_url,
+                    extract=extract,
+                )
+            else:
+                job = await data_copier.launch(
+                    storage_uri=destination_url,
+                    src_uri=source_url,
+                    dst_uri=URL("/var/storage"),
+                    extract=extract,
+                )
+            while job.status == neuro_api.JobStatus.PENDING:
+                job = await client.jobs.status(job.id)
+                await asyncio.sleep(1.0)
+            async for chunk in client.jobs.monitor(job.id):
+                if not chunk:
+                    break
+                click.echo(chunk.decode(errors="ignore"), nl=False)
+            job = await client.jobs.status(job.id)
+            if job.status == neuro_api.JobStatus.FAILED:
+                logger.error("The copy job has failed due to:")
+                logger.error(f"  Reason: {job.history.reason}")
+                logger.error(f"  Description: {job.history.description}")
+                exit_code = job.history.exit_code
+                if exit_code is None:
+                    exit_code = EX_PLATFORMERROR
+                sys.exit(exit_code)
+            else:
+                logger.info("Successfully copied data")
+    else:
+        TEMP_UNPACK_DIR.mkdir(parents=True, exist_ok=True)
+        tmp_dir_name = tempfile.mkdtemp(dir=str(TEMP_UNPACK_DIR))
+        tmp_dst_url = URL.build(path=(tmp_dir_name + "/"))
+
+        await _nonstorage_cp(source_url, tmp_dst_url)
+        source_url = tmp_dst_url
+        # for clarity
+        source_url_type = UrlType.LOCAL
+
+        # at this point source is always local
+        if extract:
+            # extract to tmp dir
+            file = Path(source_url.path)
+            if file.is_dir():
+                file = list(file.glob("*"))[0]
+            suffixes = file.suffixes
+            if suffixes[-2:] == [".tar", ".gz"] or suffixes[-1] == ".tgz":
+                command = "tar"
+                args = ["zxvf", str(file), "-C", str(tmp_dst_url)]
+            elif suffixes[-2:] == [".tar", ".bz2"] or suffixes[-1] in (".tbz2", ".tbz"):
+                command = "tar"
+                args = ["jxvf", str(file), "-C", str(tmp_dst_url)]
+            elif suffixes[-1] == ".tar":
+                command = "tar"
+                args = ["xvf", str(file), "-C", str(tmp_dst_url)]
+            elif suffixes[-1] == ".gz":
+                command = "gunzip"
```

Comment:
hmm, can't `archive.gz` be unpacked via `tar` ? My concern is to use as little side utilities as possible, if we want to be cross-platform. 

Anyways, I think, later we'll re-write everything with python's libraries to be more cross-platform.

Reply:
> hmm, can't archive.gz be unpacked via tar ? My concern is to use as little side utilities as possible, if we want to be cross-platform.
Totally agree with you, but I didn't find out a way to use tar for gzipped files (only for gzipped tar archives).

> Anyways, I think, later we'll re-write everything with python's libraries to be more cross-platform.
That's a cool idea :)